src/analysis/normed/group/add_circle.lean

```diff
@@ -0,0 +1,213 @@
+/-
+Copyright (c) 2022 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+import algebra.order.floor
+import algebra.order.to_interval_mod
+import analysis.normed.group.quotient
+
+/-!
+# The additive circle
+
+We define the additive circle `add_circle p` as the quotient `𝕜 ⧸ (ℤ ∙ p)` for some period `p : 𝕜`.
+Note that when `𝕜 = ℝ`, this is naturally a normed group; for example if `p = 1` then:
+`∥(x : add_circle 1)∥ = |x - round x|` for any `x : ℝ` (see `unit_add_circle.norm_eq`).
+
+See also `circle` and `real.angle`.
+
+## Main definitions:
+
+ * `add_circle`: the additive circle `𝕜 ⧸ (ℤ ∙ p)` for some period `p : 𝕜`
+ * `unit_add_circle`: the special case `ℝ ⧸ ℤ`
+ * `add_circle.equiv_add_circle`: the rescaling equivalence `add_circle p ≃+ add_circle q`
+ * `add_circle.equiv_Ico`: the natural equivalence `add_circle p ≃ Ico 0 p`
+ * `add_circle.norm_eq`: a characterisation of the norm on `add_circle p`
+
+## Implementation notes:
+
+Although the most important case is `𝕜 = ℝ` we wish to support other types of scalars, such as
+the rational circle `add_circle (1 : ℚ)`, and so we set things up more generally.
+
+## TODO
+
+ * Link with periodicity
+ * Measure space structure
```

Comment:
See the branch `RmodZ` for the measure space structure, particularly the theorem in `measure_theory/integral/periodic.lean` that the quotient map from `ℝ` to `add_circle` is a measure-preserving map with respect to `volume.restrict (Ioc t (t + T))` on `ℝ` and the Haar measure on `add_circle`.

Reply:
Just had a look, thanks. It certainly looks useful!